app/views/layouts/_header.html.erb

```diff
@@ -14,6 +14,17 @@
 
     <!-- Collect the nav links, forms, and other content for toggling -->
     <div class="collapse navbar-collapse js-octobox-menu" id="octobox-menu">
+      <%= form_tag root_path, method: :get, class: 'navbar-form navbar-left', enforce_utf8: false do |f| %>
+        <div class="form-group">
+          <% @text_search_hidden_field_params.each do |name, value| %>
+            <!-- filter out unwanted query params -->
+            <% if (name != 'controller') && (name != 'action') %>
+              <%= hidden_field_tag name, value  %>
+            <% end %>
```

Comment:
so the logic here is to create a hidden field for every query param we currently have, that way when the form is submitted we keep the existing, filters